shared/utils/build-remediations.py

```diff
@@ -0,0 +1,285 @@
+
+    def _set_environment(func):
+        def wrapper(self, *args):
+
+            os.environ["TEMPLATE_DIR"] = self._get_template_dir()
+            os.environ["BUILD_DIR"] = self.output_dir
+            try:
+                return func(self, *args)
+            finally:
+                os.environ["TEMPLATE_DIR"] = ""
+                os.environ["BUILD_DIR"] = ""
+        return wrapper
+
+    @_set_environment
+    def _run_script(self, script, csv_filepath):
+
+        for lang in self.langs:
+            sp = subprocess.Popen(
+                ["python", script, lang, csv_filepath],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE
+            )
+            self._subprocess_check(sp)
+
+    def _mkdir_recursive(self, path):
+
+        if not os.path.exists(path):
+            self._mkdir_recursive(os.path.realpath(os.path.dirname(path)))
+
+        if not os.path.exists(path):
+            os.mkdir(path)
+
+    @_set_environment
+    def _read_io_files_list(self, script, csv, lang, gen_input):
+
+        try:
+            if gen_input:
+                os.environ["GENERATE_INPUT_LIST"] = "true"
+            else:
+                os.environ["GENERATE_OUTPUT_LIST"] = "true"
+
+            sp = subprocess.Popen(
+                ["python", script, lang, csv],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE
+            )
+            return self._get_list_from_subprocess(sp)
+
+        finally:
+            os.environ["GENERATE_INPUT_LIST"] = ""
+            os.environ["GENERATE_OUTPUT_LIST"] = ""
+
+    def _subprocess_check(self, subprocess):
+        subprocess.wait()
+        if subprocess.returncode == 0:
+            pass
+        elif subprocess.returncode == 2:
+            pass
+        elif subprocess.returncode == 3:
+            pass
+        else:
+            raise RuntimeError("Process returned:" + subprocess.communicate()[1].decode("utf-8"))
+
+    def _get_list_from_subprocess(self, subprocess):
+
+        self._subprocess_check(subprocess)
+        text = subprocess.communicate()[0].decode("utf-8")
+        return [os.path.abspath(line) for line in text.split("\n") if line]
+
+
+    def _deduplicate(self, files):
+        return set(os.path.realpath(file) for file in files)
+
+
+if __name__ == "__main__":
+
+    builder = Builder()
+    p = argparse.ArgumentParser()
+
+    sp = p.add_subparsers(dest='cmd', help="actions")
+
+    make_sp = sp.add_parser('build', help="Build remediations")
+    make_sp.set_defaults(cmd="build")
+
+    clean_sp = sp.add_parser('clean', help="Clean remediations")
+    clean_sp.set_defaults(cmd="clean")
+
+    input_sp = sp.add_parser('input', help="Generate input list")
+    input_sp.set_defaults(cmd="input")
+
+    output_sp = sp.add_parser('output', help="Generate output list")
+    output_sp.set_defaults(cmd="output")
+
+    p.add_argument('--input',        action="store",                      help="input directory")
```

Comment:
I think we agreed to have `--input` be required but in the latest version it's optional again. IMO it's better if we explicitly pass `.` instead of having it be the default.